packages/web-components/src/components/va-checkbox-group/va-checkbox-group.tsx

```diff
@@ -0,0 +1,81 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Listen,
+  Prop,
+  h,
+} from '@stencil/core';
+
+@Component({
+  tag: 'va-checkbox-group',
+  styleUrl: 'va-checkbox-group.css',
+  shadow: true,
+})
+export class VaCheckboxGroup {
+  @Element() el: HTMLElement;
+
+  /**
+   * The text label for the checkbox group.
+   */
+  @Prop() label: string;
+
+  /**
+   * Whether or not this input field is required.
+   */
+  @Prop() required: boolean = false;
+
+  /**
+   * A string with an error message.
+   */
+  @Prop() error: string;
+
+  /**
+   * Whether or not an analytics event will be fired.
+   */
+  @Prop() enableAnalytics: boolean;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  @Listen('vaChange')
+  vaChangeHandler(event: CustomEvent): void {
+    const clickedItem = event.target as HTMLVaCheckboxElement;
+    if (this.enableAnalytics) this.fireAnalyticsEvent(clickedItem.label);
+  }
+
+  private fireAnalyticsEvent(optionLabel) {
+    this.componentLibraryAnalytics.emit({
+      componentName: 'va-checkbox-group',
+      action: 'change',
+      details: {
+        label: this.label,
+        optionLabel,
+        required: this.required,
+      },
+    });
+  }
+
+  render() {
+    return (
+      <Host role="group">
+        <legend>
+          {this.label}
+          {this.required && <span class="required-span">(*Required)</span>}
```

Comment:
The left margin is creating the illusion of a space here, but I wonder how that would present for screen readers. Could they tell that there are two separate words in `label(*Required)`?

![image](https://user-images.githubusercontent.com/2008881/150392673-aa341386-018c-4ff4-b5fd-3b63f169fcee.png)

@CourtneyChambers77 do you have any thoughts on this?

Reply:
I will test to ensure the spaces are not heard with SR

Reply:
@CourtneyChambers77 here is a link to the deployed version: https://60f9b557105290003b387cd5-xdrfwzdspu.chromatic.com/?path=/docs/components-va-checkbox-group--default
You can toggle on the `Required` field in the Story to test